src/nvim/search.c

```diff
@@ -1808,6 +1808,9 @@ pos_T *findmatchlimit(oparg_T *oap, int initc, int flags, int64_t maxtravel)
     initc = NUL;
   } else if (initc != '#' && initc != NUL) {
     find_mps_values(&initc, &findc, &backwards, true);
+    if (dir) {
+      backwards = (dir == FORWARD) ? FALSE : TRUE;
```

Comment:
```suggestion
      backwards = (dir == FORWARD) ? false : true;
```

Reply:
Thanks @zeertzjq! I applied your suggestion manually and squashed it into the original commit.